src/Control/Monad/Logic/Sequence/Internal.hs

```diff
@@ -325,6 +339,21 @@ instance Monad m => MonadLogic (SeqT m) where
       Empty -> single Nothing
       a :< t -> single (Just (a, t))
 
+-- | @choose = foldr (\a s -> pure a <|> s) empty@
+--
+-- @choose :: Monad m => [a] -> SeqT m a@
+choose :: (F.Foldable t, Monad m) => t a -> SeqT m a
+choose = F.foldr cons empty
```

Comment:
Does base have any rules for `F.foldr`? I'm tempted to say mark this `INLINEABLE` just to be safe. Same for `chooseM`.